### PR TITLE
Compute the statistics in parallel

### PR DESCRIPTION
**Pull request by micheles**
This the companion of https://github.com/gem/oq-hazardlib/pull/482 and it is needed to close https://github.com/gem/oq-engine/issues/2180.
The tests are running here: https://ci.openquake.org/job/zdevel_oq-engine/2139

### CODE REVIEW

File: openquake/calculators/classical.py [hunk 1]
```diff
@@ -345,102 +346,119 @@ def post_execute(self, pmap_by_grp_id):
...
 
 
+def build_hcurves_and_stats(pmap_by_grp, sids, pstats, rlzs_assoc, monitor):
+    """
+    :param pmap_by_grp: dictionary of probability maps by source group ID
+    :param sids: array of site IDs
+    :param pstats: instance of PmapStats
+    :param rlzs_assoc: instance of RlzsAssoc
+    :param monitor: instance of Monitor
+    """
```
Comment (ptormene) [inline openquake/calculators/classical.py]: Please add the description of the return value

File: openquake/commonlib/datastore.py [hunk 2]
```diff
@@ -216,16 +222,17 @@ def get_attr(self, key, name, default=None):
...
+    def create_dset(self, key, dtype, shape=(None,), compression=None,
+                    attrs=None):
         """
         Create a one-dimensional HDF5 dataset.
 
         :param key: name of the dataset
         :param dtype: dtype of the dataset (usually composite)
-        :param size: size of the dataset (if None, the dataset is extendable)
+        :param shape: shape of the dataset, possibly extendable
+        :param attrs: dictionary of attributes of the dataset
```
Comment (ptormene) [inline openquake/commonlib/datastore.py]: Please describe also param compression

File: openquake/commonlib/export/hazard.py [hunk 3]
```diff
@@ -230,28 +231,33 @@ def __iter__(self):
...
-def export_hazard_curves_csv(key, dest, sitecol, curves_by_imt,
-                             imtls, investigation_time=None):
+def export_hazard_csv(key, dest, sitemesh, pmap,
+                      imtls, comment):
     """
     Export the curves of the given realization into CSV.
 
     :param key: output_type and export_type
     :param dest: name of the exported file
     :param sitecol: site collection
```
Comment (ptormene) [inline openquake/commonlib/export/hazard.py]: Please replace sitecol with sitemesh also in the docstring

Comment (ptormene) [inline openquake/commonlib/export/hazard.py]: ...and add the description of imtls

File: openquake/calculators/classical.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/classical.py]: Done

File: openquake/commonlib/datastore.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/datastore.py]: Done

File: openquake/commonlib/calc.py [hunk 4]
```diff
@@ -99,6 +99,21 @@ def gen_ruptures_for_site(site, sources, maximum_distance, monitor):
             source_rupture_sites, key=operator.attrgetter('source')):
         yield src, [row.rupture for row in rows]
 
+
+# used in classical and event_based calculators
+def combine_pmaps(rlzs_assoc, results):
+    """
+    :param results: dictionary src_group_id -> probability map
```
Comment (ptormene) [inline openquake/commonlib/calc.py]: Please add to the docstring also the parameter rlzs_assoc

**Comment by ptormene** [comment]
LGTM

**Pull request merged**